python_hooks/image_tag_branch_constraint.py

```diff
@@ -0,0 +1,49 @@
+import argparse
+import ast
+import sys
+import enum
+from python_hooks.utills.ignore_check import ignore_check
+from python_hooks.utills.ignore_check import IdentifierCheck
+
+DISALLOWED_MESSAGE = "Flagged {name} in {filename}:{line}."
+
+def check_file(filename):
+    check_accumulator = []
+    with open(filename) as file:
+        tree = ast.parse(file.read(), filename=filename)
+
+    for node in ast.walk(tree):
+        if isinstance(node, ast.Call):
+            if isinstance(node.func, ast.Name) and node.func.id in ['DatabricksJobOperator', 'DatabricksImageRepo', 'DatabricksSharedOperator', 'DatabricksNotebookOperator']:
```

Comment:
nit: I don't think we need `DatabricksImageRepo` here, since that is just an enum for the ECR repo names